server/controllers/checkoutController/checkout.controller.js

```diff
@@ -0,0 +1,150 @@
+const { stripe, stripeWebhookSecret } = require('../../config');
+const stripeConnect = require('stripe')(stripe);
+const { sendEmail } = require('../../utils/mailer');
+
+module.exports.createCheckoutSession = async (req, res) => {
+  /* example req =
+    "items": [
+        {"id": "price_1LSOxMFiSX0kathO35npm8yT", "quantity": 3},
+        {"id": "price_1LSOw6FiSX0kathOSA1GkzVF", "quantity": 1}
+             ]
+          */
+  //retrieve each product and info needed for line items
+  //has to be done here to prevent people from changing the price of items they purchase etc
+  let storeItems = [];
+  for (let i = 0; i < req.body.items.length; i++){
```

Comment:
This is going to take forever to do. There has to be a better way to do this. Depending on how many items they have, they might have to wait 30 or more seconds for this for loop.

Reply:
Could you possible group them all up and then send them to stripe? or does it have to be for each item?

Reply:
Yeah, this is not very efficient. I almost left a comment saying that I hated this implementation but I think it might actually be necessary. I'll review the stripe docs and see if there is a better way on Monday. The problem is I need the price from Stripe themselves since otherwise customers can easily tamper with the price as they pay. So I need to retrieve the price object for that. But I also need to get the name of the item since I'm just working with an ID ( unless this will be given to me in the request, I assumed not and the code to get the product name could easily be removed later if this is the case ), so then I have to retrieve the product object. Then I have to create the checkout session with stripe. I'll look into if there is a more efficient way but I don't believe so unless the FE will be passing the name... still a call to stripe per item plus a call at the end for the session creation. Dang that was an essay, sorry!